src/scenes/pt.cu

```diff
@@ -183,29 +225,84 @@ CU_HD Intersection Disc::intersect(Ray const &r, Object const *obj) const {
   return {t, obj, x, normal};
 }
 
+CU_D ObjectSample Object::sample(Vec3 const &dir, TraceContext &ctx) const {
+  ObjectSample sample;
+  if (type == SPHERE) {
+    sample = sphere.sample(dir, ctx, this);
+  } else if (type == PLANE) {
+    sample = plane.sample(dir, ctx, this);
+  } else if (type == DISC) {
+    sample = disc.sample(dir, ctx, this);
+  }
+  return sample;
+}
+
+CU_D ObjectSample Sphere::sample(Vec3 const &hitx, TraceContext &ctx,
+                                 [[maybe_unused]] Object const *obj) const {
+  Vec3 p = uniformHemisphereSampling(ctx);
+  p      = onb(p, obj->tr.translation() - hitx);
```

Comment:
Should just assume center of 0 here and all that. Do the transform outside of the specific sample method.
Anyway...isn't the hemisphere facing the wrong way?